### ISSUE TITLE
klipper-helm image needs updating

### ISSUE THREAD

**Issue by brandond** [k3s-io/helm-controller#65]
The klipper-helm image has become stale:
* Based on Alpine 3.8 which went end-of-support 2020-05-01
* Built with golang 1.11 which has not been supported since 1.13 came out
* Packages old versions of helm v2/v3

**Comment by cjellick** [comment]
@brandond are you trying to get this through and into 1.19.1? Might make sense to wait to the next 1.19 release to update it in k3s to reduce test scope (there will be plenty.

**Comment by cjellick** [comment]
cc @davidnuzik

**Comment by davidnuzik** [comment]
It looks like it's in now so we have to test it regardless? QA will need some high level information on how this should be tested - what areas to work on. @brandond cc: @cjellick

**Comment by brandond** [comment]
I haven't updated the helm-controller version in k3s or rke2 yet; that can wait if we'd like but since we're working on fixing vulns in images it should probably be done sooner rather than later.

**Comment by davidnuzik** [comment]
Thanks - as discussed in standup we'll standby for now and not update the version in k3s or rke2 just yet. Let's get v1.19.1 k3s out then we can update this post that release. cc: @cjellick

**Comment by brandond** [comment]
The work here is already done, but absent an issue in the k3s repo I'm going to keep this around until we pull through into there.

**Comment by davidnuzik** [comment]
Is this ready to test or need to pull through into k3s?

**Comment by davidnuzik** [comment]
Caution: no milestone because in a different org right now, needs to be transferred to k3s project once we are able.

**Comment by dweomer** [comment]
Would like to pull through to the new org. Waiting on an issue with integrations for helm-controller, though.

**Comment by davidnuzik** [comment]
Needs PR to bump this in K3s. Until then QA isn't able to test.

**Comment by brandond** [comment]
@dweomer is there any way we can run CI on this now without waiting for CNCF infra, or is it pretty much stuck until then?

**Comment by dweomer** [comment]
> @dweomer is there any way we can run CI on this now without waiting for CNCF infra, or is it pretty much stuck until then?

I believe that CI is stuck until I get it working due to github+drone integration limitations.

**Comment by davidnuzik** [comment]
**QA:**
- Ensure charts deploy as expected.
- Inspect pods for helm-install-* jobs; should see klipper-helm container and this should have the new imager version bump. 
From [PR](https://github.com/k3s-io/helm-controller/pull/66): "rancher/klipper-helm:v0.3.1"

**Comment by rancher-max** [comment]
I believe this needs to be pulled into k3s and rke2 still. When I check for the klipper image in rke2 helm pods, I still see using v0.3.0:

```
$ k get -n kube-system pod/helm-install-rke2-canal-7c7xl -o yaml | grep -i klipper
    image: rancher/klipper-helm:v0.3.0
    image: docker.io/rancher/klipper-helm:v0.3.0
    imageID: docker.io/rancher/klipper-helm@sha256:9cab2b9230d0454933de576b4e768cda8e0f1395cd5aca86ab649fc1d3be47bc
```

**Comment by ieugen** [comment]
@rancher-max : I've also noticed the old klipper-helm image . 
Any way to update the klipper-helm version manually until a fix is pushed?
Not entirely sure but does it make sense to specify the klipper-helm image and version to use when installing apps - in the Chart manifest !?!

**Comment by ieugen** [comment]
You can use a newer image via `spec.jobImage`, docs PR https://github.com/rancher/docs/pull/2873 . 
As for updating the actual image itself, that is another issue.

**Comment by rancher-max** [comment]
#### Validated in k3s v1.20.0+k3s2 and rke2 v1.19.5+rke2r1

This will continue in the v1.20.x line and above of k3s and the v1.19.x line and above of rke2.

- For k3s:
```
$ kubectl get -n kube-system pod/helm-install-traefik-hfsp2 -o yaml | grep -i klipper
    image: rancher/klipper-helm:v0.3.2
    image: docker.io/rancher/klipper-helm:v0.3.2
    imageID: docker.io/rancher/klipper-helm@sha256:33f8d2b6961b76a10826d75194d8678a0e3ad4d715abf8e7370ac552a91eb0f0
```
- for rke2:
```
$ k get -n kube-system pod/helm-install-rke2-canal-qrjn4 -o yaml | grep -i klipper
    image: rancher/klipper-helm:v0.3.2
    image: docker.io/rancher/klipper-helm:v0.3.2
    imageID: docker.io/rancher/klipper-helm@sha256:33f8d2b6961b76a10826d75194d8678a0e3ad4d715abf8e7370ac552a91eb0f0
```
- All pods install correctly 

@davidnuzik Do we want a backport into other lines of rke2 and k3s? Leaving this open pending that potential.

**Comment by davidnuzik** [comment]
@brandond to open backports at his next availability. cc: @rancher-max
